### DISCUSSION TITLE
Getting Historical changes

### DISCUSSION THREAD

**Discussion by pnewell** [supabase/supabase#30601] [Questions] [answered]
I am playing with the Realtime Postgres Changes feature and wondering if it is possible to get historical changes? Like if I want to get every version for a specific row? I'd like to use this for syncing but makes it hard that there is no way to get older changes

**Comment by GaryAustin1** [comment, marked as answer]
No.  You would have to come up with something on your own for that.

You could for instance have a trigger for all operations that inserts to a logging table every operation and data.  Then realtime could monitor that logging table.   It could also fetch older changes in that table by using a timestamp.   But you would likely need a cron task to clean up entries after a certain period of time depending on the number of changes.
